pkg/gktest/suite.go

```diff
@@ -1,31 +1,197 @@
 package gktest
 
-import "io/fs"
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
 
-// Suite defines a set of TestCases which all use the same ConstraintTemplate
-// and Constraint.
+	"github.com/open-policy-agent/frameworks/constraint/pkg/apis"
+	templatesv1 "github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/core/templates"
+	"gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// scheme stores the k8s resource types we can instantiate as Templates.
+var scheme = runtime.NewScheme()
+
+func init() {
+	_ = apis.AddToScheme(scheme)
+}
+
+// Suite defines a set of Constraint tests.
 type Suite struct {
-	TestCases []TestCase
+	metav1.ObjectMeta
+
+	// Constraints is a list of Template&Constraint pairs, with tests to run on
+	// each.
+	Constraints []constraint
```

Comment:
This could get confusing as it's overloading the terminology.

For instance, is a constraint name:

* `Suite.Constraint[_].constraint` (in which case it is the file name)
* The `metadata.name` of the constraint

?

Unless we are forcing those two values to be equal. Do we want that requirement? Can we enforce it?

Also, are we then coupling this schema too tightly to constraints such that it will be harder to integrate mutation in the future?

Reply:
Let's discuss - I've set up a meeting.

Reply:
Resolved by using Suite/Test/Case as the test name hierarchy.